test/cluster/test_logstor_repair.py

```
@@ -0,0 +1,1387 @@
+    by_pk: dict[int, list[Any]] = {}
+    for row in rows:
+        by_pk.setdefault(row.pk, []).append(row)
+    return {pk: _record_from_rows(pk_rows) for pk, pk_rows in by_pk.items()}
+
+
+async def read_all_records(cql: Any, hosts: list[Any], table: str) -> list[dict[int, Record]]:
+    """Read the records stored on every host, one dict per host, in `hosts` order."""
+    return [await read_records(cql, host, table) for host in hosts]
+
+
+def read_locally(cql: Any, host: Any, query: str) -> list[Any]:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `cql`

(ANN401)